src/sentry/static/sentry/app/components/events/meta/metaData.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import isNil from 'lodash/isNil';
+
+import {getMeta} from 'app/components/events/meta/metaProxy';
+import ErrorBoundary from 'app/components/errorBoundary';
+import {Meta} from 'app/types';
+
+type Props<Values, K extends Extract<keyof Values, string>> = {
+  object: Values;
+  prop: K;
+  /**
+   * Render prop that is called with these args:
+   *  value: The actual value,
+   *  meta: metadata object if it exists, otherwise undefined,
+   */
+  children: (value: Values[K], meta?: Meta) => React.ReactNode;
+  required: boolean;
+};
+
+/**
+ * Retrieves metadata from an object (object should be a proxy that
+ * has been decorated using `app/components/events/meta/metaProxy/withMeta`
+ */
+const MetaData = <Values extends {}>({
```

Comment:
Will making this component generic require updating where it is used? I'm not well versed in components with generics for props.

Reply:
no it won't be necessary Mark :)